core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -5445,29 +5455,30 @@ targetNamespace, castNonNull(catalogReader.unwrap(Prepare.CatalogReader.class)),
       throw newValidationError(createTable, RESOURCE.createTableRequiresAsQuery());
     }
 
-    final SqlValidatorScope queryScope = requireNonNull(scopes.get(queryNode));
-
     //Note, this can either a row expression or a query expression with an optional ORDER BY
-    if (queryNode instanceof SqlSelect) {
-      final SqlSelect sqlSelect = (SqlSelect) queryNode;
-      validateSelect(sqlSelect, unknownType);
-    } else {
-      validateQuery(queryNode, queryScope, unknownType);
-    }
-
-    final SqlValidatorNamespace queryNS = getNamespaceOrThrow(queryNode);
-    final DdlNamespace createTableNS = (DdlNamespace) getNamespaceOrThrow(createTable);
+    //We're not currently handling the row expression case.
+
+    SqlValidatorScope createTableScope = this.getCreateTableScope(createTable);
+    // In order to be sufficiently general to the input of Create table,
+    // We have to validate this expression in the overall scope of the create table node,
+    // Since the associated query node
```

Comment:
Can you just say something like "we have to validate in the overall scope of the create table node in order to be sufficiently general to the input of "create table as", since the associated query node (in relation to the create table node) may not be a selectScope or any one type of scope"

Reply:
done